thermal-control-design.md

```diff
@@ -0,0 +1,272 @@
+		"fan_speed_when_suspend": 60%
+	    },
+	    "fan_absence": {
+		"action": {
+		    "thermal_control_algorithm": "disable",
+		    "fan_speed": 100%,
+		    "led_color": "red"
+		}
+	    },
+	    "psu_absence": {
+		"action": {
+		    "thermal_control_algorithm": "disable",
+		    "fan_speed": 100%,
+		    "led_color": "red"
+		}
+	    },
+	    "all_fan_failed": {
+	        "action": {
+		    "shutdown_switch": true
+		 }
+	    }
+	}
+
+In this configuration, thermal control algorithm will run on this device; in fan absence situation, the fan speed need to be set to 100%, the thermal control algorithm will be suspended and fan status led shall be set to red ; in psu absence situation, thermal control algorithm will be suspend, fan speed will be set to 100% and psu status led shall be set to red.
+
+During daemon start, this configuration json file will be loaded and parsed, daemon will handle the thermal control algorithm run and fan speed set when predefined policy meet.
+
+## 4. CLI show command for temperature and fan design
+
+### 4.1 New CLI show command for temperature
+
+ adding a new sub command to the "show platform":
+ 
+	admin@sonic# show platform ?
+	Usage: show platform [OPTIONS] COMMAND [ARGS]...
+
+	  Show platform-specific hardware info
+
+	Options:
+	  -?, -h, --help  Show this message and exit.
+
+	Commands:
+	  mlnx         Mellanox platform specific configuration...
+	  psustatus    Show PSU status information
+	  summary      Show hardware platform information
+	  syseeprom    Show system EEPROM information
+	  temperature  Show device temperature information
+	  
+out put of the new CLI
+
+    admin@sonic# show platform temperature
+    NAME Temperature       Timestamp       High Threshold   Low Threshold    Critical High Threshold   Critical Low Threshold   Warning Status
+    ---- -----------  ------------------   ---------------  --------------   ------------------------ ------------------------ ----------------
+    CPU      85        20191112 09:38:16        110              -10                 120                        -20                   false
+    ASIC     75        20191112 09:38:16        100               0                  110                        -10                   false
+
+An option '--major' provided by this CLI to only print out major device temp, if don't want show all of sensor temperatures.
+Major devices are CPU pack, cpu cores, ASIC and optical modules.
+
+### 4.2 New show CLI for fan status
+
+We don't have a CLI for fan status getting yet, new CLI for fan status could be like below, it's adding a new sub command to the "show platform":
+
+	admin@sonic# show platform ?
+	Usage: show platform [OPTIONS] COMMAND [ARGS]...
+
+	  Show platform-specific hardware info
+
+	Options:
+	  -?, -h, --help  Show this message and exit.
+
+	Commands:
+	  fanstatus  Show fan status information
+	  mlnx       Mellanox platform specific configuration...
+	  psustatus  Show PSU status information
```

Comment:
psustatus may be enhanced to dump PSU FAN values as well..

Reply:
psu fan should be covered by fan cli, not sure whether we want to add PSU fan info here.

Reply:
From a modularity perspective, PSU FANs are added under PsuBase : so ideally, PSU FANs should be dumped as part of psustatus if at all.

Reply:
@jleveque what's your view on show PSU fan info as part of PSU show cli? Since we have a PSU daemon, if go this way we may consider to extend psud to get it's fan info?

Reply:
I see how it could be helpful to display PSU fan info along with status, voltage, etc. I think it is worth exploring extending the PSU daemon.